skrub/_table_vectorizer.py

```diff
@@ -149,141 +149,7 @@ def _replace_missing_in_cat_col(ser: pd.Series, value: str = "missing") -> pd.Se
-                    new_transformers.append((name_split, trans, [col]))
-                    new_transformer_to_input_indices[name_split] = [trans_to_mapping]
-            else:
-                new_transformers.append((name, trans, cols))
-                new_transformer_to_input_indices[name] = transformers_to_input_indices[
-                    name
-                ]
-
-    return new_transformers, new_transformer_to_input_indices
-
-
-def _merge_transformers(
-    transformers: list[tuple[str, Transformer, list[str]]],
-    is_fitted: bool,
-    transformer_to_input_indices: dict[str, list[int]] | None = None,
-) -> tuple[list[tuple[str, Transformer, list[str]]], dict[str, list[int]]]:
-    """
-    Merge splitted transformers into a single transformer.
-
-    Parameters
-    ----------
-    transformers : list of 3-tuples (str, Transformer or str, list of str)
-        The collection of transformers to merge, as tuples of
-        (name, transformer, column).
-    is_fitted : bool
-        Whether the transformers are fitted or not, i.e. whether the
-        `transformers` argument corresponds to the `self.transformers_` attribute
-        (when True) or the `self.transformers` attribute (when False).
-    transformer_to_input_indices : dict of str to list of int, optional
-        The mapping of transformer names to the indices of the columns they were
-        fitted on. Should correspond to the `self._transformer_to_input_indices`
-        attribute. Only used when `is_fitted` is True.
-    """
-    new_transformers = []
-    new_transformer_to_input_indices = {} if is_fitted else transformer_to_input_indices
-    base_names = pd.unique(
-        pd.Series([name.split("_split_")[0] for name, _, _ in transformers])
-    )
-
-    for base_name in base_names:
-        # merge all transformers with the same base name
-        transformers_base_name, names, columns = [], [], []
-        for name, trans, cols in transformers:
-            if name.startswith(base_name):
-                columns.extend(cols)
-                transformers_base_name.append(trans)
-                names.append(name)
-
-        new_trans = transformers_base_name[0]
-        if not is_fitted:
-            if isinstance(new_trans, TransformerMixin):
-                new_trans = clone(new_trans)
-        else:
-            if len(transformers_base_name) > 1:
-                # merge transformers
-                new_trans = new_trans.__class__._merge(transformers_base_name)
-            new_transformer_to_input_indices[base_name] = list(
-                chain.from_iterable(
-                    [transformer_to_input_indices[name] for name in names]
-                )
-            )
-        new_transformers.append((base_name, new_trans, columns))
-
-    return new_transformers, new_transformer_to_input_indices
-
-
-class TableVectorizer(ColumnTransformer):
+class TableVectorizer(TransformerMixin, _BaseComposition):
```

Comment:
Yep. But we don't have any test so I am not sure of the behaviour now.

Reply:
I checked that all the tests still pass if we use `BaseEstimator` instead of  `_BaseComposition`. It would be good if we could derive from the `BaseEstimator` which is meant to be subclassed, rather than the private `_BaseComposition`. but if you think this needs more investigation I can open a separate issue

Reply:
`_BaseComposition` is indeed private but this is a rather stable piece of code that was not really change over the passed year. It would make sense in scikit-learn to expose it as a developer base class. I think that it is safe enough to use it here.

Reply:
cool thanks for explaining!